contracts/account-history-contract/src/types/account_snapshot_generator.rs

```diff
@@ -654,165 +653,86 @@ impl AccountSnapshotGenerator {
-        let denom_uusdc = denom_usdc_entry.entry.denom;
-        let usdc_display_denom = denom_usdc_entry.entry.display_name;
-
-        let denom_uelys = ElysDenom::Elys.as_str().to_string();
-        let denom_ueden = ElysDenom::Eden.as_str().to_string();
-        let denom_uedenb = ElysDenom::EdenBoost.as_str().to_string();
-
-        let usdc_oracle_price = querier.get_oracle_price(
-            usdc_display_denom.clone(),
-            ElysDenom::AnySource.as_str().to_string(),
-            0,
-        )?;
-        let usdc_price = usdc_oracle_price
-            .price
-            .price
-            .checked_div(Decimal::from_atomics(Uint128::new(1000000), 0).unwrap())
-            .unwrap_or_default();
+        let rewards_response = querier.get_all_program_rewards(address.to_string())?;
 
-        let mut balance_rewards: Vec<BalanceReward> = vec![];
-        let mut rewards = Reward {
-            usdc_usd: Decimal::zero(),
-            eden_usd: Decimal::zero(),
-            eden_boost: Uint128::zero(),
-            other_usd: Decimal::zero(),
-            total_usd: Decimal::zero(),
-        };
+        // Concatenate all staking reward vectors into one
+        let all_staking_rewards: Vec<DecCoin> = rewards_response
+            .usdc_staking_rewards
+            .iter()
+            .chain(&rewards_response.elys_staking_rewards)
+            .chain(&rewards_response.eden_staking_rewards)
+            .chain(&rewards_response.edenb_staking_rewards)
+            .cloned()
+            .collect();
 
-        match commitments.commitments.rewards_unclaimed {
-            Some(rewards_unclaimed) => {
-                for reward in rewards_unclaimed {
-                    // uusdc
-                    if reward.denom == denom_uusdc {
-                        let usdc_rewards =
-                            Decimal::from_atomics(reward.amount, 0).unwrap_or_default();
-                        let rewards_in_usd = usdc_rewards.checked_mul(usdc_price)?;
-
-                        balance_rewards.push(BalanceReward {
-                            asset: denom_usdc_entry.entry.base_denom.clone(),
-                            amount: reward.amount,
-                            usd_amount: Some(rewards_in_usd),
-                        });
-
-                        rewards.usdc_usd = rewards_in_usd;
-                        rewards.total_usd = rewards
-                            .total_usd
-                            .checked_add(rewards.usdc_usd)
-                            .unwrap_or_default();
-
-                        continue;
-                    }
+        // Accumulate amounts for each denomination
+        let mut denom_amounts: HashMap<String, Decimal256> = HashMap::new();
+        for coin in &all_staking_rewards {
+            let amount = denom_amounts
+                .entry(coin.denom.clone())
+                .or_insert_with(Decimal256::zero);
+            *amount += coin.amount;
```

Comment:
no checked add is used

in this way we are not able to check the overflow condition